scarlet/observation.py

```diff
@@ -231,3 +253,146 @@ def get_scene(self, scene):
             msg = "get_scene is currently only supported when the observation frame matches the scene"
             raise NotImplementedError(msg)
         return self.images
+
+
+
+class Combination(Scene):
+
+
+    def __init__(self, images, psfs=None, weights=None, wcs = None, filtercurve=None, padding=3, target_psf = None, structure = None):
+        super().__init__(images.shape, wcs=wcs, psfs=psfs, filtercurve=filtercurve)
+
+        self.images = np.array(images)
+        self.padding = padding
+
+        if weights is not None:
+            self._weights = np.array(weights)
+        else:
+            self._weights = 1
+
+        if target_psf is not None:
+            self.target_psf = target_psf
+        else:
+            self.target_psf = target_psf
+        self.structure = structure
+
+
+    def match(self, scene):
+
+        if self.wcs == None:
+            raise TypeError('WCS is actually mandatory, please provide one (tbdiscussed)')
+        if self.psfs is not None:
+            #Get pixel coordinates in each frame.
+            mask, over_lr, over_hr  = resampling.match_patches(scene.shape, self.shape,scene.wcs, self.wcs)
+            self._over_lr = over_lr
+            self._over_hr = over_hr
+            self._mask = mask
+
+
+            #Compute diff kernel at hr
+
+            whr = scene.wcs
+            wlr = self.wcs
+
+            #Reference PSF
+            if self.target_psf is None:
+
+                if np.size(np.shape(scene.psfs)) ==2:
+                    _target = scene.psfs
+                    _shape = scene.shape
+                elif np.size(np.shape(scene.psfs)) ==3:
+                    _target = scene.psfs[0,:,:]
+                    _shape = scene.shape[1:]
+                else:
+                    raise ValueError('Wrong dimensions for psfs. psfs should be of dimensions n1xn2 or nbxn1xn2')
+            else:
+                _target = self.target_psf
+            # 3) compute obs.psf in the frame of scene
+            resconv_op = []
+
+            for _psf in self.psfs:
+
+
+                #Computes spatially matching observation and target psfs. The observation psf is also resampled to the scene's resolution
+
+                new_target, observed_psf = resampling.match_psfs(_target, _psf, whr, wlr)
+                #Computes the diff kernel in Fourier
+                target_fft = np.fft.fft2(np.fft.ifftshift(new_target))
+                observed_fft = np.fft.fft2(np.fft.ifftshift(observed_psf))
+                kernel_fft = np.zeros(target_fft.shape)
+                kernel_fft[target_fft != 0] = observed_fft[target_fft != 0] / target_fft[target_fft != 0]
```

Comment:
THis solved errors when applying the algorithm, big time!

Reply:
Remember, here observed and target psfs are not the same at all! So zero values in target_fft will create infinitely high frequencies. The result, is that without discarding null values, the ringing is made worse. Also, it introduces a  shift. I think I know why, but I'll take a look at that.

Reply:
This is OK if the target_fft has zeros, i.e. no power at some frequencies, which means it's well-resolved. I'd suggest to rewrite as 
```
sel = target_fft != 0
kernel_fft[sel] = observed_fft[sel] / target_fft[sel]
```
to avoid having to do the selection three times.

Reply:
Exactly, well resolved cases should be handled properly. If if not well resolved, it doesn't hurt.